PLAN-849 Replace session auth throughout backend code

lastminutediorama opened the pull request:
This changes how we handle authentication on the backend to follow django_rest_framework patterns, rather than just django patterns.

The vast majority of the edits here fit one of these categories:
- adding `@api_view` decorators and appropriate REST verbs to all of our endpoints
- switching tests to use force_authenticate instead of force_login
- ensuring that we send properly transformed JSON strings as payloads for non-GET requests in each test, since the rest framework is more strict about this
- returning a consistent JSON error and 401 when a user isn't authenticated
- cleaning up some of the http error codes that I encountered which were clearly misleading (I know there are more of these to cleanup, but that seemed like tech debt outside of this scope)

We may want to do an extra set of QA tests before deploying this, since it touches so much of the code.

pflopez reviewed (approved):
Code seems ok to me (might want someone else to double check) but did functional testing of this and all seems good!
- Was able to log in and out
- Was able to create a new account
- Getting data (plans/scenarios) worked, as well as creating new.
- Change password, name and last name worked

pflopez commented on src/planscape/planning/views.py:
```diff
@@ -170,9 +157,9 @@ def delete_planning_area(request: HttpRequest) -> HttpResponse:
-        user = _get_user(request)
-        if user is None:
-            raise ValueError("User must be logged in.")
+        user = request.user
```
Could we move this to a function? its repeated on all the methods

lastminutediorama commented:
Ok, for transparency, here are some manual tests I ran locally:
- with currently logged-in user:
    - removed sessionid cookie
    - navigated throughout the existing Prod "smoke tests" as listed in Jira
 saw no issues, no new sessionid cookie was set.
 
 Then,
     -created a new user
     - changed password
     -  reset password 
     - navigated throughout the existing Prod "smoke tests" as listed in Jira
 saw no issues, no new sessionid cookie was set. 

I'd still like to test the other endpoints as changed in this PR, but I wanted to make this note before I went to lunch and forgot what I did :D
